common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model.questbook;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.managers.CoreManager;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wynn.event.QuestBookReloadedEvent;
+import com.wynntils.wynn.event.WorldStateEvent;
+import com.wynntils.wynn.model.WorldStateManager;
+import com.wynntils.wynn.model.container.ContainerContent;
+import com.wynntils.wynn.model.container.ScriptedContainerQuery;
+import com.wynntils.wynn.utils.ContainerUtils;
+import com.wynntils.wynn.utils.InventoryUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class QuestBookManager extends CoreManager {
+    private static final Pattern QUEST_NAME_MATCHER = Pattern.compile("^§.§l([^֎À]*)[֎À]+ (§e\\[Tracked\\])?$");
+    private static final int NEXT_PAGE_SLOT = 8;
+
+    private static List<QuestInfo> quests = List.of();
+    private static List<QuestInfo> newQuests;
+
+    public static void init() {}
+
+    @SubscribeEvent
+    public static void onQuestsReloaded(QuestBookReloadedEvent event) {
+        McUtils.player().sendMessage(new TextComponent("Quest Book scanned.").withStyle(ChatFormatting.GRAY), null);
+    }
+
+    @SubscribeEvent
```

Comment:
You might want to lower priority, so the quest book rescanning is always scheduled after the player info scan.

Reply:
This also causes a bug. (Scripted container queries dont have a queue and overlap each other..) I think we do not need to do this anymore, tracking does not depend on quest book anymore.

Reply:
I could have sworn I added a queue to the scripted container queries, just since they did have a chance to overlap and collide? If I did not do that, it is definitely needed.

Reply:
I don't think there is a queue (as there is also no way for a query to signal completion.

Reply:
```
    public static void runQuery(ContainerQueryStep firstStep) {
        if (currentStep != null) {
            // Only add if it is not already enqueued
            if (queuedQueries.stream()
                    .filter(query -> query.getName().equals(firstStepName))
                    .findAny()
                    .isEmpty()) {
                queuedQueries.add(firstStep);
            }
            return;
        }
```

Reply:
So yes, there is a queue, and yes, the container query manager knows when it is done (that is when the query do not return a "next step").

Reply:
Nice, though, I will still remove this since this is not needed anymore.

Reply:
I'm not sure I understood why this is not needed anymore. :) I'll see if I can figure it out and come back and answer my own question. Otherwise feel free to enlighten me. :)

Reply:
We do not need to analyse quests on world join. Legacy needed the quest book always tracked so quest overlay worked. We only need quest book tracking when the custom quest book GUI is open. So, we don't need this here, and we can query the quest book when we open the GUI.

Reply:
I think I am following now, yes. :) Wynncraft quest tracking is reset each time you leave a world, right?